crypto/ec/ecp_secp384r1.c

```diff
@@ -0,0 +1,19594 @@
+                                          fiat_secp384r1_uint1 *out2,
+                                          fiat_secp384r1_uint1 arg1,
+                                          uint64_t arg2, uint64_t arg3) {
+    fiat_secp384r1_int128 x1;
+    fiat_secp384r1_int1 x2;
+    uint64_t x3;
+    x1 = ((arg2 - (fiat_secp384r1_int128)arg1) - arg3);
+    x2 = (fiat_secp384r1_int1)(x1 >> 64);
+    x3 = (uint64_t)(x1 & UINT64_C(0xffffffffffffffff));
+    *out1 = x3;
+    *out2 = (fiat_secp384r1_uint1)(0x0 - x2);
+}
+
+/*
+ * The function fiat_secp384r1_mulx_u64 is a multiplication, returning the full double-width result.
+ * Postconditions:
+ *   out1 = (arg1 * arg2) mod 2^64
+ *   out2 = ⌊arg1 * arg2 / 2^64⌋
+ *
+ * Input Bounds:
+ *   arg1: [0x0 ~> 0xffffffffffffffff]
+ *   arg2: [0x0 ~> 0xffffffffffffffff]
+ * Output Bounds:
+ *   out1: [0x0 ~> 0xffffffffffffffff]
+ *   out2: [0x0 ~> 0xffffffffffffffff]
+ */
+static void fiat_secp384r1_mulx_u64(uint64_t *out1, uint64_t *out2,
+                                    uint64_t arg1, uint64_t arg2) {
+    fiat_secp384r1_uint128 x1;
+    uint64_t x2;
+    uint64_t x3;
+    x1 = ((fiat_secp384r1_uint128)arg1 * arg2);
+    x2 = (uint64_t)(x1 & UINT64_C(0xffffffffffffffff));
+    x3 = (uint64_t)(x1 >> 64);
+    *out1 = x2;
+    *out2 = x3;
+}
+
+/*
+ * The function fiat_secp384r1_cmovznz_u64 is a single-word conditional move.
+ * Postconditions:
+ *   out1 = (if arg1 = 0 then arg2 else arg3)
+ *
+ * Input Bounds:
+ *   arg1: [0x0 ~> 0x1]
+ *   arg2: [0x0 ~> 0xffffffffffffffff]
+ *   arg3: [0x0 ~> 0xffffffffffffffff]
+ * Output Bounds:
+ *   out1: [0x0 ~> 0xffffffffffffffff]
+ */
+static void fiat_secp384r1_cmovznz_u64(uint64_t *out1,
+                                       fiat_secp384r1_uint1 arg1, uint64_t arg2,
+                                       uint64_t arg3) {
+    fiat_secp384r1_uint1 x1;
+    uint64_t x2;
+    uint64_t x3;
+    x1 = (!(!arg1));
+    x2 = ((fiat_secp384r1_int1)(0x0 - x1) & UINT64_C(0xffffffffffffffff));
+    x3 = ((x2 & arg3) | ((~x2) & arg2));
+    *out1 = x3;
+}
+
+/*
+ * The function fiat_secp384r1_mul multiplies two field elements in the Montgomery domain.
+ * Preconditions:
+ *   0 ≤ eval arg1 < m
+ *   0 ≤ eval arg2 < m
+ * Postconditions:
+ *   eval (from_montgomery out1) mod m = (eval (from_montgomery arg1) * eval (from_montgomery arg2)) mod m
+ *   0 ≤ eval out1 < m
+ *
+ * Input Bounds:
+ *   arg1: [[0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff]]
+ *   arg2: [[0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff]]
+ * Output Bounds:
+ *   out1: [[0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff]]
```

Comment:
This line looks too long for our coding style.

Reply:
I was wrapping before but @romen gave me the `clang-format` option to preserve this one. Which I think makes sense.